### PR TITLE
issue #164: Crash at ngspice output file parsing

### PR DESCRIPTION
**Pull request by Michi1994**
The solution is: define the new expected end of simulation as the last output time of ngspice.

### CODE REVIEW

**Comment by Michi1994** [comment]
Maybe a message would be cool so that the user knows that ngspice did not simulate to the end?

File: src/engines/ngspice.c [hunk 1]
```diff
@@ -447,7 +451,7 @@ static void ngspice_start (OreganoEngine *self)
 	                              &priv->child_error,  // STDERR
 	                              &e)) {
 		// Add a watch for process status
-		g_child_watch_add_full (priv->child_pid, G_PRIORITY_HIGH_IDLE,
+		g_child_watch_add_full (G_PRIORITY_HIGH_IDLE, priv->child_pid,
```
Comment (drahnr) [inline src/engines/ngspice.c]: Holy s***, how did this go unnoticed?!?

**Comment by drahnr** [comment]
For the future, it would be cool if you could do a `git pull --rebase` on the master branch instead of criss-cross merging :)

**Comment by Michi1994** [comment]
Maybe a message would be cool so that the user knows that ngspice did not simulate to the end?

**Comment by drahnr** [comment]
Yep a `g_warning` would be nice for now, with the big pr that is stil pending a menu re-design this will be easier to handle.

**Comment by Michi1994** [comment]
What do you mean by "pr"?

**Comment by drahnr** [comment]
pr == pullrequest

**Comment by drahnr** [comment]
Alternatively I created a log-view.c and log.c to be able to display chains of error events to the user, so you would need to get the pointer to the log struct and call `log_append_error` or `log_append`.

**Pull request merged**